test/test_unit_suffixes.py

```diff
@@ -0,0 +1,148 @@
+"""
+Test unit suffix functionality for length-based options.
+
+This module contains tests for the unit suffix parsing feature added to Filtlong.
+To run them, execute `python3 -m unittest test.test_unit_suffixes` from the
+root Filtlong directory.
+"""
+
+import unittest
+import os
+import subprocess
+
+
+class TestUnitSuffixes(unittest.TestCase):
+
+    def run_command(self, command):
+        binary_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'bin', 'filtlong')
+        input_path = os.path.join(os.path.dirname(__file__), 'test_sort.fastq')
+        assembly_reference = os.path.join(os.path.dirname(__file__), 'test_reference.fasta')
+
+        command = command.replace('filtlong', binary_path)
+        command = command.replace('INPUT', input_path)
+        command = command.replace('ASSEMBLY', assembly_reference)
+
+        output_name = 'TEMP_' + str(os.getpid())
+        command = command.replace('OUTPUT', output_name)
+        try:
+            self.output_file = [x for x in command.split() if output_name in x][0]
+        except IndexError:
+            self.output_file = ''
+        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+        _, err = p.communicate()
+        return err.decode(), p.returncode
+
+    def tearDown(self):
+        if os.path.isfile(self.output_file):
+            os.remove(self.output_file)
+
+    def test_target_bases_k_suffix(self):
+        """Test that k suffix works for target_bases."""
+        console_out, return_code = self.run_command('filtlong --target_bases 10k INPUT > OUTPUT.fastq')
+        self.assertEqual(return_code, 0)
+        self.assertTrue('target: 10000 bp' in console_out)
```

Comment:
the way this comes out in logs is with comma separation. i.e. target: 700,000,000 bp